modules/misc/nixgl.nix

```diff
@@ -39,26 +84,58 @@ in {
           buildCommand = ''
             set -eo pipefail
 
-            ${
-            # Heavily inspired by https://stackoverflow.com/a/68523368/6259505
-            pkgs.lib.concatStringsSep "\n" (map (outputName: ''
-              echo "Copying output ${outputName}"
-              set -x
-              cp -rs --no-preserve=mode "${
-                pkg.${outputName}
-              }" "''$${outputName}"
-              set +x
-            '') (old.outputs or [ "out" ]))}
+            ${# Heavily inspired by https://stackoverflow.com/a/68523368/6259505
+              pkgs.lib.concatStringsSep "\n" (map (outputName: ''
+                echo "Copying output ${outputName}"
+                set -x
+                cp -rs --no-preserve=mode "${pkg.${outputName}}" "''$${outputName}"
+                set +x
+              '') (old.outputs or [ "out" ]))}
 
-            rm -rf $out/bin/*
             shopt -s nullglob # Prevent loop from running if no files
+
+            rm -rf $out/bin/*
             for file in ${pkg.out}/bin/*; do
               echo "#!${pkgs.bash}/bin/bash" > "$out/bin/$(basename $file)"
-              echo "exec -a \"\$0\" ${cfg.prefix} $file \"\$@\"" >> "$out/bin/$(basename $file)"
+              echo "${environment}" >> "$out/bin/$(basename $file)"
+              echo "exec -a \"\$0\" ${getWrapperExe vendor} $file \"\$@\"" >> "$out/bin/$(basename $file)"
               chmod +x "$out/bin/$(basename $file)"
             done
+
+            # If .desktop files refer to the old package, replace the references
+            for dsk in "$out/share/applications"/*.desktop ; do
+              if ! grep "${pkg.out}" "$dsk" > /dev/null; then
+                continue
+              fi
+              src="$(readlink "$dsk")"
+              rm "$dsk"
+              sed "s|${pkg.out}|$out|g" "$src" > "$dsk"
+            done
+
             shopt -u nullglob # Revert nullglob back to its normal default state
           '';
         }));
+  in {
+    lib.nixGL.wrapMesa = makePackageWrapper "Intel" "";
+    lib.nixGL.wrapNvidia = makePackageWrapper "Nvidia" nvOffloadEnv;
```

Comment:
Shouldn't nvOffloadEnv inclusion here depend on the value of `cfg.nvidiaOffload`, like it does for the shell wrapper?